### DISCUSSION TITLE
Information needed regarding rabbitmq-auth-backend-http

### DISCUSSION THREAD

**Discussion by shifty21** [rabbitmq/rabbitmq-server#2661] [Other] [answered]
Hi,

I am new to rabbitmq and I have created a Go-based HTTP backend with reference to sample HTTP backends provided in the rabbitmq codebase. Although things seem to be working fine as far as publishing and consuming messages are concerned.

I could really use some help in getting some clarification for a few things!

1. From admin user I am not able to see which users have permission to vhosts (I assume this is because rabbitmq doesn't have the list of users and it's fetched on a request basis from the backend). Is there a way to get this information?
2. For a particular user with vhost, the queue matrices in UI are not visible. Is this natural or I am missing something?
3. I am still able to add a user via the admin account and assign vhosts to it. Given I am using custom-backend this should not work, right?

The admin have (administrator) tag, while a normal user have (monitoring) tag
References 

https://www.rabbitmq.com/management.html
https://github.com/rabbitmq/rabbitmq-server/tree/master/deps/rabbitmq_auth_backend_http
https://rawcdn.githack.com/rabbitmq/rabbitmq-management/v3.8.9/priv/www/api/index.html

**Comment by michaelklishin** [comment, marked as answer]
1. Your understanding is correct.
2. I'm afraid I don't know what "queue matrices" are. If you mean "queue metrics" then those can be disabled in the management UI, e.g. when [Prometheus and Grafana](https://www.rabbitmq.com/prometheus.html) are used. We cannot suggest much given the amount of information provided. Consider inspecting your [effective configuration](https://www.rabbitmq.com/configure.html#verify-configuration-effective-configuration) and see your browser's Web console for any possible exceptions or JS files that might have been failed to download.
3. Using a custom backend does not prevent you from adding users to the internal database. Whether this will be consulted depends on the configuration of the backends.

**Reply by shifty21** [reply]
@michaelklishin  Thanks for the quick response
2. Yes, it's queue metrics. It's disabled by default, I missed it thanks for point it out.
3. In this case the request is still authorized by the custom backend which doesn't have information about rabbitmq's internal users. Isn't kind of misleading, since you can only create the users in rabbitmq and cant use them to perform anything?

BTW, you can close this discussion. Thanks again for the response.
